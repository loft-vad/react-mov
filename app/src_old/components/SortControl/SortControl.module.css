.wrapper {
  display: flex;
  justify-content: space-between;
  gap: 30px;
  align-items: center;
}
.wrapper > div {
  font-size: 16px;
  line-height: 20px;
  letter-spacing: 0.888889px;
  text-transform: uppercase;
  color: #FFFFFF;
  mix-blend-mode: normal;
  opacity: 0.6;
}
.wrapper select {
  border: none;
  background: none;
  font-weight: 500;
  font-size: 16px;
  line-height: 20px;
  letter-spacing: 0.888889px;
  text-transform: uppercase;
  color: var(--white);
}
.wrapper select option {
  color: var(--black);
}

/*# sourceMappingURL=SortControl.module.css.map */
