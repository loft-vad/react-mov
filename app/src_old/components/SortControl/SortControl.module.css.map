{"version":3,"sourceRoot":"","sources":["SortControl.module.scss"],"names":[],"mappings":"AAAA;EACE;EACA;EACA;EACA;;AAEA;EACE;EACA;EACA;EACA;EACA;EACA;EACA;;AAGF;EACE;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;AAEA;EACE","file":"SortControl.module.css"}