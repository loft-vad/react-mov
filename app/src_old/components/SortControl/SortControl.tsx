import React from "react";
import styles from "./SortControl.module.css";

export interface sortByValue {
  id: number;
  name: string;
  value?: string;
}

export interface SortByProps {
  values: sortByValue[];
  selected: sortByValue["name"];
  onSelect: (value: sortByValue["name"]) => void;
}

const SortControl: React.FC<SortByProps> = ({ values, selected, onSelect }) => {
  return (
    <div className={styles.wrapper}>
      <div>Sort by</div>
      <select
        defaultValue={selected}
        onChange={(e) => onSelect(e.target.value)}
        data-testid="select"
      >
        {values.map(({ id, name, value = "" }) => (
          <option key={id} value={value} data-testid="option">
            {name}
          </option>
        ))}
      </select>
    </div>
  );
};

export default SortControl;
