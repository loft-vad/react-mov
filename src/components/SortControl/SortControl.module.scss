.wrapper {
  display: flex;
  justify-content: space-between;
  gap: 30px;
  align-items: center;

  &>div {
    font-size: 16px;
    line-height: 20px;
    letter-spacing: 0.888889px;
    text-transform: uppercase;
    color: #FFFFFF;
    mix-blend-mode: normal;
    opacity: 0.6;
  }

  select {
    border: none;
    background: none;
    font-weight: 500;
    font-size: 16px;
    line-height: 20px;
    letter-spacing: 0.888889px;
    text-transform: uppercase;
    color: var(--white);

    option {
      color: var(--black)
    }
  }
}