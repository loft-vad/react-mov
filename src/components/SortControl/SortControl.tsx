import React from "react";
import styles from "./SortControl.module.scss";

export interface sortByValue {
  id: number;
  name: string;
}

export interface SortByProps {
  values: sortByValue[];
  selected: sortByValue["name"];
  onSelect: (value: sortByValue["name"]) => void;
}

const SortControl: React.FC<SortByProps> = ({ values, selected, onSelect }) => {
  return (
    <div className={styles.wrapper}>
      <select>
        {values.map(({ id, name }) => (
          <option
            key={id}
            value={name}
            onClick={() => onSelect(name)}
            selected={selected === name ? true : false}
          >
            {name}
          </option>
        ))}
      </select>
    </div>
  );
};

export default SortControl;
