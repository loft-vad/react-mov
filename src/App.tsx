import React from "react";
import styles from "./App.module.scss";
import Counter from "./components/Counter/Counter";
import SearchForm from "./components/SearchForm/SearchForm";
import GenreSelect, { genre } from "./components/GenreSelect/GenreSelect";
import MovieTile from "./components/MovieTile/MovieTile";
import MovieDetails, { MovieFull } from "./components/MovieDetails/MovieDetails";
import moviesDb from "./data/movies";
import SortControl, { sortByValue } from "./components/SortControl/SortControl";

export const sortBy: sortByValue[] = [
  { id: 1, name: "Release Date" },
  { id: 2, name: "Title" },
];

const App: React.FC = () => {
  const genres: genre[] = [
    { id: 1, name: "All" },
    { id: 2, name: "Documentary" },
    { id: 3, name: "Comedy" },
    { id: 4, name: "Horror" },
    { id: 5, name: "Crime" },
  ];

  const movies: MovieFull[] = moviesDb;

  const onSearch = (searchString: string) => {
    console.log("searchString ", searchString);
  };

  const handleGenreSelection = (name: string) => {
    console.log(name, " clicked");
  };

  const handleMovieTileClick = () => {
    console.log("Movie tile clicked");
  };

  const handleSortByClick = (name: string) => {
    console.log("Sort By item cliecked: ", name);
  };

  return (
    <div className={styles.contentWrapper}>
      <header>
        <div className={styles.wrapper}>
          <SearchForm onSearch={onSearch} />
        </div>
      </header>
      <div>
        <MovieDetails movieData={movies[0]} />
      </div>
      <div className={styles.content}>
        <div className={styles.searchFilter}>
          <GenreSelect genres={genres} onSelect={handleGenreSelection} />
          <SortControl selected={sortBy[0].name} values={sortBy} onSelect={handleSortByClick} />
        </div>
        <div className={styles.moviesList}>
          {movies.map((movie) => (
            <MovieTile movieData={movie} onClickHandler={handleMovieTileClick} />
          ))}
        </div>
        <div className={styles.counter}>
          <Counter initialValue={0} />
        </div>
      </div>
    </div>
  );
};

export default App;
