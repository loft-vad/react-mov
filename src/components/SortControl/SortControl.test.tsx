import React from "react";

import SortControl from "./SortControl";
import { fireEvent, render } from "@testing-library/react";
import { sortBy } from "../../App";

describe("GenreSelect component", () => {
  it("should match snapshot", () => {
    const wrapper = render(
      <SortControl values={sortBy} selected={sortBy[0].name} onSelect={() => {}} />,
    );

    expect(wrapper.container.firstChild).toMatchSnapshot();
  });
  it("renders all genres passed in props", () => {
    const wrapper = render(
      <SortControl values={sortBy} selected={sortBy[0].name} onSelect={() => {}} />,
    );
    const renderedItems = wrapper.getAllByTestId("genre").map((item) => item.textContent);
    const inputItems = sortBy.map((item) => item.name);
    expect(renderedItems).toEqual(inputItems);
  });
  it("highlights a selected genre passed in props", () => {
    const activeItem = sortBy[0].name;
    const wrapper = render(
      <SortControl values={sortBy} selected={sortBy[0].name} onSelect={() => {}} />,
    );
    const activeItemElement = wrapper.container.querySelector(".active");
    expect(activeItemElement?.textContent).toEqual(activeItem);
  });
  it('after a click event on a genre button component calls "onChange" callback and passes correct genre in arguments', () => {
    const callback = jest.fn();
    const wrapper = render(
      <SortControl values={sortBy} selected={sortBy[0].name} onSelect={() => {}} />,
    );
    const button = wrapper.container.querySelector("button");
    fireEvent.click(button!);
    expect(callback).toHaveBeenCalledWith(sortBy[0].name);
  });
});
