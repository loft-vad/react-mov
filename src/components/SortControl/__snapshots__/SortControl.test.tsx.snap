// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreSelect component should match snapshot 1`] = `
<div
  class="wrapper"
>
  <select>
    <option
      value="Release Date"
    >
      Release Date
    </option>
    <option
      value="Title"
    >
      Title
    </option>
  </select>
</div>
`;
